src/wasm/asm-wasm-builder.cc

```diff
@@ -453,26 +574,108 @@ class AsmWasmBuilderImpl : public AstVisitor {
   }                                                                   \
   break
 
+  Expression* GetLeft(BinaryOperation* expr) {
```

Comment:
Basically, these are for the type coercion statements. Intish to int is |0. Conversion to int is ~~ which translates to ((X xor -1) xor -1) and conversion to double is +x which translates to x * 1.0. In all the cases left one is the useful construct. Someone might have similar things on the other side but that's not something we are optimizing for now. At this point it is only type coercions that would be all over the place.